apps/activity/views.py

```diff
@@ -1,5 +1,196 @@
+
+    @action(methods=['get'], detail=False, serializer_class=GetTaskSerializer, url_path="tasks")
+    def tasks(self, request, *args, **kwargs):
+        return super().list(request, *args, **kwargs)
+
+    @action(methods=['post'], detail=True, serializer_class=None, url_path="add-task")
+    def add_task(self, *args, **kwargs):
+        task = get_object_or_404(Task, pk=kwargs.get("pk"))
+        plan = self.request.user.plan
+        plan.tasks.add(task)
+        plan.save()
+        plan_serializer = PlanSerializer(plan)
+        serialized_plan = plan_serializer.data
+        return Response(serialized_plan, status.HTTP_200_OK)
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(methods=['delete'], detail=True, serializer_class=None, url_path="delete-task")
+    def delete_task(self, *args, **kwargs):
+        task = get_object_or_404(Task, pk=kwargs.get("pk"))
+        plan = self.request.user.plan
+        plan.tasks.remove(task)
+        plan.save()
+        plan_serializer = PlanSerializer(plan)
+        serialized_plan = plan_serializer.data
+        return Response(serialized_plan, status.HTTP_200_OK)
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(methods=['put'], detail=True, serializer_class=None, url_path="start-task")
+    def start_task(self, *args, **kwargs):
+        task = Task.objects.get(id=kwargs.get("pk"))
+        plan = self.request.user.plan
+        task_exists = plan.tasks.filter(id=task.id).exists()
+        if task_exists:
+            plan.started_task = task
+            plan.start_task = timezone.now()
+            plan.save()
+        plan_serializer = PlanSerializer(plan)
+        serialized_plan = plan_serializer.data
+        return Response(serialized_plan, status.HTTP_200_OK)
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(methods=['put'], detail=False, serializer_class=None, url_path="continue-task")
+    def continue_task(self, *args, **kwargs):
+        plan = self.request.user.plan
+        task = plan.started_task
+        task_duration = timedelta(hours=task.duration.hour, minutes=task.duration.minute)
+        time_elapsed = plan.end_task - plan.start_task
+        if time_elapsed < task_duration:
+            plan.end_task = None
+            plan.save()
+        plan_serializer = PlanSerializer(plan)
+        serialized_plan = plan_serializer.data
+        return Response(serialized_plan, status.HTTP_200_OK)
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(methods=['put'], detail=False, serializer_class=None, url_path="cancel-task")
+    def cancel_task(self, *args, **kwargs):
+        plan = self.request.user.plan
+        plan.start_task = None
+        plan.end_task = None
+        plan.started_task = None
+        plan.save()
+        plan_serializer = PlanSerializer(plan)
+        serialized_plan = plan_serializer.data
+        return Response(serialized_plan, status.HTTP_200_OK)
```

Comment:
PlanSerializer(plan).data in asa mod vei scapa de 2 rinduri de mai sus, ca avantaj obtinem mai putin cod